### ISSUE TITLE
Patched: Unnecessary errors during Postfix installation

### ISSUE THREAD

**Issue by bogdro** [OpenMandrivaAssociation/distribution#2789]
**OpenMandriva version**: 4.3

**Describe the bug**:
When installing Postfix (+the required minimum dependencies), the following messages show up in the console:
```
chown: cannot access '/usr/lib64/postfix/postfix-cdb.so': No such file or directory
ldd: ./postfix-cdb.so: No such file or directory
ldd: ./postfix-ldap.so: No such file or directory
ldd: ./postfix-mysql.so: No such file or directory
ldd: ./postfix-pcre.so: No such file or directory
ldd: ./postfix-pgsql.so: No such file or directory
ldd: ./postfix-sdbm.so: No such file or directory
ldd: ./postfix-sqlite.so: No such file or directory
```
No idea where the `chown` comes from, but `ldd` is called by `/usr/sbin/postfix-chroot.sh`.
The script runs `ldd` on all libraries found in the configuration file no matter if they exist.

**Steps to reproduce**:
Install Postfix (run `dnf install postfix`).

**Observed behavior**:
The above errors are shown in the terminal.

**Expected behavior**:
No errors are shown.

**Additional comment**:
The attached patch should fix the issue. You can also do something like `[[ -e $i ]] && ldd $i` etc. if you wish.
[postfix-chroot.sh.patch.gz](https://github.com/OpenMandrivaAssociation/distribution/files/8324890/postfix-chroot.sh.patch.gz)

**Logs and screenshots if relevant**

**Comment by ZeroAbility** [comment]
It doesn't appear the patch was ever applied. Is this still an issue?

**Comment by bogdro** [comment]
Hello.
From what I see - yes.
I've installed 3.8.3-1-omv2390 as a test and got:
- coredumps for `/usr/bin/postmap /etc/postfix/virtual`, `/usr/bin/postmap /etc/postfix/domains` and `/usr/bin/postfix set-permissions` (but that may be the fault of my system),
- the same "No such file or directory" errors,
- a message that `sendmail-command` should be linked to `/usr/sbin/sendmail`,
- a message that `%post(postfix-3.8.3-1.x86_64)` failed with exit code 2.

**Comment by ZeroAbility** [comment]
> I've installed 3.8.3-1-omv2390 as a test and got:

Could you try a newer version of OMLx, please?

**Comment by bogdro** [comment]
There is no newer version.
Even though the RPM package has "omv2390" in the name, it is in the OMLx 5.0's repository: https://abf-downloads.openmandriva.org/5.0/repository/x86_64/main/release/. Probably not a good naming scheme (randomly picking, I see many other 2390's, many 4050's, many 4090's and even 22090's, but very few omv5000's).
The version I originally tried with was 3.6.5, now I tried with 3.8.3 - the newest OMLx has in the stable repositories (and the newest version on Postfix's site is 3.10.1).

**Comment by ZeroAbility** [comment]
Looks like the newest version should be available in Rock 6.0, but is available currently in `cooker` and a newer version exists in ROME.

**Comment by bogdro** [comment]
I've downloaded `postfix-3.10.0-1-omv2590.x86_64.rpm` from https://abf-downloads.openmandriva.org/cooker/repository/x86_64/main/release/ and confirm that the issue with `ldd` still exists (no more `chown`, though, but I didn't find it in the original installation scripts either).

**Comment by ZeroAbility** [comment]
It looks to me like the goal of doing the `ldd` step is to perform necessary steps against db connectors and other features the user *may* install. Most people at the sys admin level are going to have a db in mind that they want to use with `postfix`. I guess the question is, does the "error" still occur with libraries that are actually there? You would need to install the package for the db you want to use.

**Comment by bogdro** [comment]
Probably the error won't appear if I have the library, but why should I install all of them? If I have one DB, I wouldn't install 2 more.

**Comment by ZeroAbility** [comment]
> You would need to install the package for the db you want to use.

Do the rest of the errors prevent the package from being installed after you have installed the db you want to use?

**Comment by bogdro** [comment]
No, the errors don't prevent the installation at all, whether I install the DB I want or not.
I just wanted to point out the unnecessary (and perhaps confusing) messages during installation.

**Comment by ZeroAbility** [comment]
@berolinux can probably chime in on this, but the patch LGTM. It probably won't be a priority since the package install still works minus the conspam, and we are trying to get 6.0 done.

**Comment by bogdro** [comment]
Yeah, yeah, no problem. Should have mentioned that the installation works.

**Comment by ZeroAbility** [comment]
The spec repo is here. 

https://github.com/OpenMandrivaAssociation/postfix

Please feel free to evaluate it and provide a PR for any changes that your patch would fix.
